CI.hs

```diff
@@ -258,8 +244,6 @@ buildDists
 
     -- Make and extract an sdist of ghc-lib.
     cmd "cd ghc && git checkout ."
-    appendFile "ghc/hadrian/stack.yaml" $ unlines ["ghc-options:","  \"$everything\": -O0 -j"]
-    -- Feedback on the compiler used for ghc-lib-gen.
```

Comment:
We didn't add any extra-deps here but we seem to add them globally with `applyPatchHadrianStackYaml` now. Is this intended?

Reply:
The extra-deps were added a little bit further down: https://github.com/digital-asset/ghc-lib/blob/6be486b61f07c022402d632e161ab081c837a74b/CI.hs#L247-L249. This block is now correctly moved to ghc-lib-gen and an additional extra dep added (which is temporary while the upstream fix isn't landed).

Reply:
The line you're referring to is a bit further up. Let me rephrase what I find confusing: Before this change we had tw locations where we wrote `ghc-options` to `stack.yaml` and only location where we wrote `extra-deps`. Now, we have one location where we write both. What happened to the case where we need to write the `ghc-options` but not the `extra-deps`?

Reply:
Before in CI.hs, we unconditionally wrote ghc-options and wrote extra-deps conditionally on flavor ghc-master in two consecutive statements. These two statements have been moved to ghc-lib-gen and the interior of the function `applyPatchHadrianStackYaml`. The function will not write extra-deps if the flavor is not ghc-master just as it were before.

At the risk of confusing things further but in the interests of completeness, there is one further change that is being overlooked: CI.hs used to patch hadrian.cabal exactly once. This was a (then not,  but now important) error. It should happen twice because in-between generating ghc-lib-parser.cabal and ghc-lib.cabal, the changes to hadrian/hadrian.cabal are reverted. Now, `Main.hs` arranges to patch  hadrian's cabal  on every execution of ghc-lib-gen regardless of whether the target is ghc-lib-parser or ghc-lib.

Reply:
In any case, the key point is that the two statements that were in CI.hs are exactly next two each other in ghc-lib-gen now as they were before https://github.com/shayne-fletcher/ghc-lib/blob/03e3d68df6fcd132fe7a9b836c9cd90b0616cde0/ghc-lib-gen/src/Ghclibgen.hs#L584-L594.

Reply:
@hurryabit  Ah! I see what you are saying. Yes, we were only writing the extra-deps once before. That is what I am wittering on about here https://github.com/digital-asset/ghc-lib/pull/245#discussion_r492125958.  We are agreed that this was not correct and this PR incidentally fixes that.

Reply:
That explains it. Thanks a lot.